core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/DataColumn.kt

```diff
@@ -74,47 +89,134 @@ public interface DataColumn<out T> : BaseColumn<T> {
+         * @param values the values to represent each row in the column
+         * @param suggestedType optional suggested type for values. Default is [TypeSuggestion.Infer].
+         *   See [TypeSuggestion] for more information.
+         * @param nullable optionally you can specify whether [values] contains nulls, if `null` it is inferred.
+         */
         public fun <T> createWithTypeInference(
             name: String,
             values: List<T>,
+            suggestedType: TypeSuggestion = TypeSuggestion.Infer,
             nullable: Boolean? = null,
-        ): DataColumn<T> = guessColumnType(name, values, nullable = nullable)
+        ): DataColumn<T> =
+            createColumnGuessingType(
+                name = name,
+                values = values,
+                suggestedType = suggestedType,
+                nullable = nullable,
+            )
 
-        public fun <T> create(
+        /**
+         * Calls [createColumnGroup], [createFrameColumn], or [createValueColumn] based on
+         * [type].
+         *
+         * This may be unsafe but is more efficient than [createWithTypeInference].
+         *
+         * Be careful; Values in [values] are NOT checked to adhere to the given [type], nor
+         * do we check whether there are unexpected nulls among the values.
+         *
+         * It's recommended to use [createValueColumn], [createColumnGroup], and [createFrameColumn] instead.
+         *
+         * @param name the name of the column
+         * @param values the values to represent each row in the column
+         * @param type the (unchecked) common type of [values]
+         * @param infer in case a [ValueColumn] is created, this controls how/whether types need to be inferred
+         */
+        public fun <T> createUnsafe(
             name: String,
             values: List<T>,
             type: KType,
             infer: Infer = Infer.None,
         ): DataColumn<T> =
-            when (type.toColumnKind()) {
+            when (type.toColumnKind()) { // AnyFrame -> Frame, AnyRow? -> Group, else -> Value
                 ColumnKind.Value -> createValueColumn(name, values, type, infer)
+
                 ColumnKind.Group -> createColumnGroup(name, (values as List<AnyRow?>).concat()).asDataColumn().cast()
+
                 ColumnKind.Frame -> createFrameColumn(name, values as List<AnyFrame>).asDataColumn().cast()
             }
 
-        public inline fun <reified T> create(name: String, values: List<T>, infer: Infer = Infer.None): DataColumn<T> =
-            create(name, values, typeOf<T>(), infer)
+        /**
+         * Calls [createColumnGroup], [createFrameColumn], or [createValueColumn] based on
+         * type [T].
+         *
+         * This is generally safe, as [T] can be inferred, and more efficient than [createWithTypeInference].
+         *
+         * Be careful when casting occurs; Values in [values] are NOT checked to adhere to the given/inferred type [T],
```

Comment:
It is unsafe because of type? For me createUnsafe creates weird reference to something like `sun.misc.Unsafe`. Could we rename it somehow to the `createWithoutTypeChecking` or` createAndFailFast`

Reply:
>createAndFailFast

haha, another title could be `createRaw`, but `createWithoutTypeChecking` might also be clear. It's unsafe in the sense that it solely uses the given type `T` to decide which type of column to instantiate. It doesn't check any of the values, which makes it fast, but, well, unsafe.

Reply:
Hmm, on the other hand, then we can have `DataColumn.createWithoutTypeInference(..., infer = Infer.Type)`. That's gonna be confusing haha. We need a same that says it will choose createValueColumn, createColumnGroup, or createFrameColumn based on the type T

Reply:
Alright, I think I'm gonna go with:
- createColumnByInference
- createColumnByType